backend/studytimeboard/routes.py

```diff
@@ -23,11 +23,32 @@
 from flask import Flask, request
 from flask_login import login_user, logout_user
 
+from functools import wraps
+
 # internal utils
 from . import app, dbapi, logger
 from .constant import *
 from .app_utils import *
 
+# database proxies
+from .utils.mongodb_client import users_collection
+# utils for hasing password and jwt encode/decode
+from .utils.bcrypt_utils import hash_password, validate_password
+from .utils.jwt_utils import jwt_encode, jwt_decode_identity, jwt_verify
+
+def login_required():
```

Comment:
please move this to app_utils.py.

app_utils is a place to put all not-route function. (will be refactored, but so far please put it into the app_utils)